### ISSUE TITLE
Can I propose the change based on experiment?

### ISSUE THREAD

**Issue by kouohhashi** [SeanNaren/deepspeech.pytorch#532]
Hi, 
I've encountered **nan loss problem** when I trained my model with spec_augment.

Since it happens after some epochs, it's not easy to find the root cause.
But symptom is that after some epochs,  loss becomes non.

<img width="873" alt="Screen Shot 2020-05-25 at 9 34 03 AM" src="https://user-images.githubusercontent.com/5446294/82768798-2903e880-9e6c-11ea-97f0-604156aa10c8.png">

my training scripts is like ...
```
python train.py --rnn-type lstm --hidden-size 1024 --hidden-layers 5  --train-manifest train.csv --val-manifest valid.csv --speed-volume-perturb --spec-augment --epochs 100 --num-workers 8 --cuda  --learning-anneal 1.01 --batch-size 32 --no-sortaGrad --opt-level O1 --loss-scale 1 --id sotuu_b7 --checkpoint --save-folder snapshots --model-path snapshots/deepspeech_final.pth
```

But somehow I think I found a way to prevent non loss problem.
Solution is making the range of point_to_warp a bit smaller.

line 56 of  time_warp function in spec_augment.py
from ```point_to_warp = horizontal_line_at_ctr[random.randrange(W, spec_len-W)]``` to ```point_to_warp = horizontal_line_at_ctr[random.randrange(W+1, spec_len-W-1)]```

<img width="857" alt="Screen Shot 2020-05-25 at 9 32 55 AM" src="https://user-images.githubusercontent.com/5446294/82769035-9106fe80-9e6d-11ea-8cae-03d670685af7.png">

My question is this.
Can I propose the change even if I don't know why this change prevent nan loss problem?

I mean this change is not  based on theory but based on experiment.

I'm pretty sure this small change can make training with spec_augument robust but I  couldn't explain why this change works....

**Comment by SeanNaren** [comment]
Now that is a fun little find, I can't see any clear reason why this would help do you mind debugging a bit further? It seems like adding a `-1` would prevent an overflow problem but I can't see how it would overflow...

I'll also check other implementations of spec augment to see if this behaviour is different

**Comment by kouohhashi** [comment]
Seems like  nan loss issue is nothing to do with spec augment or data augmentation.
Sorry for bothering you....

Because input data itself is always ok as far as I checked, but sometimes gradients are updated to nan, probably because of cuda/apex bug or settings. After reinstalling cuda and apex, I have not encountered the nan loss yet. Anyway I'll wait  until 100 epochs completed and report the results.

**Comment by SeanNaren** [comment]
thanks @kouohhashi i'll close this for now, but hopefully Pytorch 1.6 comes out and we can say bye to Apex...
